File: leetcode/twoSum.js
var twoSum = function (nums, target) {
  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      if (nums[i] + nums[j] === target) {
        return [i, j];
        break;
      }
    }
  }
  return [];
};

const nums = [3, 2, 4];
const target = 6;

console.log(twoSum(nums, target));

File: 2025/leetcode /twoSum.py
from typing import List

# class Solution:
#     def twoSum(self, nums: List[int], target: int) -> List[int]:
#       output = []

#       i = 0
#       j = i + 1

#       numCount = len(nums)

#       for i in range(0, numCount):
#         nums_i = nums[i]
#         for j in range(i+1, numCount):
#           nums_j = nums[j]
#           summed = nums_i + nums_j 

#           if summed == target:
#             output = [i, j]

#       return output

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
      output = []
      hashmap = dict()
      i = 0

      numCount = len(nums)

      for i in range(0, numCount):
        nums_i = nums[i]
        if nums_i in hashmap:
          return [hashmap[nums_i], i]
        diff = target - nums_i
        
        hashmap[diff] = i


      return output


nums = [3,3]
target = 6

sol = Solution()
print(sol.twoSum(nums, target))